### PR TITLE
fix: make NPG error code decoding lenient

### PR DESCRIPTION
**Pull request by pietro-tota**
#### List of Changes
Make NPG error responses error codes lenient

#### Motivation and Context
This modification is needed since NPG error codes list is not finite and can dynamically be changed

#### How Has This Been Tested?
Junit tests

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/pagopa/ecommerce/commons/client/NpgClient.java [hunk 1]
```diff
@@ -839,8 +686,15 @@ private NpgResponseException exceptionToNpgResponseException(
...
-                        .map(error -> GatewayError.valueOf(error.getCode())).toList();
+                errors = Optional
+                        .ofNullable(responseErrors)
+                        .map(
+                                errorDetails -> errorDetails
+                                        .stream()
+                                        .map(ErrorsInnerDto::getCode)
+                                        .toList()
+                        )
+                        .orElse(List.of());
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/ecommerce/commons/client/NpgClient.java:697]: Easier to read maybe?

```suggestion
                errors = Optional
                        .ofNullable(responseErrors)
                        .orElse(List.of())
                        .stream()
                        .map(ErrorsInnerDto::getCode)
                        .toList();
```

File: src/test/java/it/pagopa/ecommerce/commons/client/NpgClientTests.java [hunk 2]
```diff
@@ -813,19 +815,112 @@ void shouldPropagateErrorCodesWhileGetState() throws JsonProcessingException {
...
+                .expectErrorMatches(
+                        e -> {
+                            if (e instanceof NpgResponseException npgResponseException) {
+                                assertTrue(npgResponseException.getErrors().isEmpty());
+                                assertEquals(HttpStatus.BAD_REQUEST, npgResponseException.getStatusCode().get());
+                                return true;
+                            } else {
+                                fail("Unexpected exception received %s".formatted(e));
+                                return false;
+                            }
```
Comment (giovanniberti) [inline src/test/java/it/pagopa/ecommerce/commons/client/NpgClientTests.java:867]: Are there issues with casting directly to `NpgResponseException`?

**Review by giovanniberti — DISMISSED** [review]
Great work! 👏

File: src/test/java/it/pagopa/ecommerce/commons/client/NpgClientTests.java (hunk 2 above)
Comment (pietro-tota) [reply src/test/java/it/pagopa/ecommerce/commons/client/NpgClientTests.java:867]: No I just wanted here to make test failure with a custom message, I will refactor it to cast it directly

File: src/main/java/it/pagopa/ecommerce/commons/client/NpgClient.java (hunk 1 above)
Comment (pietro-tota) [reply src/main/java/it/pagopa/ecommerce/commons/client/NpgClient.java]: Fixed with https://github.com/pagopa/pagopa-ecommerce-commons/pull/178/commits/000cab563e84831ac0eef2ca29fde9df363f7095

File: src/test/java/it/pagopa/ecommerce/commons/client/NpgClientTests.java (hunk 2 above)
Comment (pietro-tota) [reply src/test/java/it/pagopa/ecommerce/commons/client/NpgClientTests.java]: Fixed with https://github.com/pagopa/pagopa-ecommerce-commons/pull/178/commits/000cab563e84831ac0eef2ca29fde9df363f7095

**Review by giovanniberti — APPROVED** [review]

**Pull request merged**